src/document/utils/list-optimized-to-print-fill.js

```diff
@@ -0,0 +1,41 @@
+/** @typedef {import("../builders.js").Doc} Doc */
+
+/**
+ * This class is a wrapper around an array that provides optimized subset of methods
+ * - constructor(array) - creates a new instance of the class with the given array with O(1) complexity
+ * - get length() - returns the length of the list with O(1) complexity
+ * - at(index) - returns the element at the given index with O(1) complexity
+ * - slice(start) - returns a new instance of the class with the underlying array sliced from the given index with O(1) complexity without changing the original list
+ */
+class ListOptimizedToPrintFill {
```

Comment:
The alternative term "Slice" is clearer than "List" here to programmers who are familiar with some other langs like Rust and Go. Here I use "List" because "Slice" can be confusing with Array.prototype.slice() in JavaScript, that creates copied new array.
Ask me to rename it to slice if it's more understandable to you.

Reply:
If we use "Slice", OptimizedFor... looks to be unnecessary.